RELEASE-NOTES.md

```diff
@@ -10,6 +10,11 @@ The Axom project release numbers follow [Semantic Versioning](http://semver.org/
 ## [Unreleased] - Release date yyyy-mm-dd
 
 ### Added
+- Added the ability to specify a user-supplied allocator ID to use with the
```

Comment:
To be clear, this is an Umpire allocator ID, correct? If so, it should be stated explicitly here. If not Umpire, what does the allocator ID refer to?

Reply:
Yes, this is an umpire allocator ID -- I will clarify.

- [ ] Clarify allocator ID.

Reply:
Fixed.